File: src/Api.Model/DivisionFileModel.cs
using System.ComponentModel.DataAnnotations;

namespace Api.Models
{
    /// <summary>
    /// Describes division that is saved in file. User for JSON conversions.
    /// </summary>
    public class DivisionFileModel
    {
        /// <summary>
        /// Name of the division. For example "Senior Wome" or "Mixed doubles"
        /// </summary>
        [Required]
        public string Name { get; set; } = null!;

        /// <summary>
        /// Competitors participating in this division.
        /// </summary>
        [Required]
        public CompetitorPositionFileModel[] Items { get; set; } = null!;
    }
}

File: frontend/src/services/openapi/models/DivisionFileModel.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { CompetitorPositionFileModel } from './CompetitorPositionFileModel';

/**
 * Describes division that is saved in file. User for JSON conversions.
 */
export type DivisionFileModel = {
    /**
     * Name of the division.
     */
    name: string;
    /**
     * Competitors participating in this division.
     */
    items: Array<CompetitorPositionFileModel>;
}
